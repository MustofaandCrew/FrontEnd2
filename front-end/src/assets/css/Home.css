/* Landing Page */
.buttonCategories {
    background: transparent;
    border: none;
    color: white;
    padding: 0px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 16px;
    margin: 0px;
    cursor: pointer;
  }
  
  .imgCarousel {
    height: 35.25rem;
    width: 120rem;
  }
  
  /* Extra small devices (phones, 600px and down) */
  @media only screen and (max-width: 600px){
    /* ini untuk tampilan handphone */
    .imgCarousel {
      height: 10rem;
      width: 120rem;
    } 
  }
  
  /* Small devices (portrait tablets and large phones, 600px and up) */
  @media only screen and (min-width: 600px){
    /* ini untuk tampilan tablet */
    .imgCarousel {
      height: 10rem;
      width: 120rem;
    }
  }
  
  /* Medium devices (landscape tablets, 768px and up) */
  @media only screen and (min-width: 768px){
    .imgCarousel {
      height: 20rem;
      width: 120rem;
    }
  }
  
  /* Large devices (Laptops/desktop, 992px and up) */
  @media only screen and (min-width: 992px){
    /* ini untuk tampilan laptop */
    .imgCarousel {
      height: 32.25rem;
      width: 120rem;
    }
  }
  
  /* Extra Large devices (Large laptops and destops, 120px and up) */
  @media only screen and (min-width: 1200px){
  }