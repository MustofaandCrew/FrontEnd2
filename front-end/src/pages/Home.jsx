import React from "react";
import Banner from "../assets/images/Banner.png";
import Jordan from "../assets/images/AirJordan.jpg";
import CatBaby from "../assets/images/CatBaby.png";
import CatGadgets from "../assets/images/CatGadgets.png";
import CatFurniture from "../assets/images/CatFurniture.png";
import CatSneakers from "../assets/images/CatSneakers.png";
import CatMakeUp from "../assets/images/CatMakeUp.png";
import CatTools from "../assets/images/CatTools.png";

import Card from "../components/Card";

import "../assets/css/Home.css";
import "../assets/css/Card.css";
import "../assets/css/Navbar.css";

// Dummy Product Images
import NewProducts from "../assets/images/NewProducts.jpg";
import NewProducts2 from "../assets/images/NewProducts2.jpg";
import NewProducts3 from "../assets/images/NewProducts3.jpg";
import LayoutNavbarFooter from "../components/LayoutNavbarFooter";

export default function Home() {
    return (
        <LayoutNavbarFooter>
            <div className="container container-fluid">

                {/* Carousel */}
                <div id="carouselExampleInterval" className="carousel slide" data-bs-ride="carousel">
                    <div className="carousel-inner">
                        <div className="carousel-item active" data-bs-interval="2000">
                            <img src={Jordan} className="d-block w-100 imgCarousel" alt="img" />
                        </div>
                        <div className="carousel-item" data-bs-interval="2000">
                            <img src={Banner} className="d-block w-100 imgCarousel" alt="img" />
                        </div>
                        <div className="carousel-item">
                            <img src={Banner} className="d-block w-100 imgCarousel" alt="img" />
                        </div>
                    </div>
                    <button className="carousel-control-prev" type="button" data-bs-target="#carouselExampleInterval" data-bs-slide="prev">
                        <span className="carousel-control-prev-icon" aria-hidden="true"></span>
                        <span className="visually-hidden">Previous</span>
                    </button>
                    <button className="carousel-control-next" type="button" data-bs-target="#carouselExampleInterval" data-bs-slide="next">
                        <span className="carousel-control-next-icon" aria-hidden="true"></span>
                        <span className="visually-hidden">Next</span>
                    </button>
                </div>

                {/* Card */}
                <div className="d-flex flex-row flex-wrap justify-content-start mt-5 mb-2">
                    <h4>Trend Categories</h4>
                </div>
                <div className="d-flex flex-row flex-wrap justify-content-around">
                    <button className="buttonCategories">
                        <div className="cardStyle">
                            <img src={CatGadgets} className="card-img-top" alt="..." />
                        </div>
                    </button>
                    <button className="buttonCategories">
                        <div className="cardStyle">
                            <img src={CatFurniture} className="card-img-top" alt="..." />
                        </div>
                    </button>
                    <button className="buttonCategories">
                        <div className="cardStyle">
                            <img src={CatMakeUp} className="card-img-top" alt="..." />
                        </div>
                    </button>
                    <button className="buttonCategories">
                        <div className="cardStyle">
                            <img src={CatSneakers} className="card-img-top" alt="..." />
                        </div>
                    </button>
                    <button className="buttonCategories">
                        <div className="cardStyle">
                            <img src={CatTools} className="card-img-top" alt="..." />
                        </div>
                    </button>
                    <button className="buttonCategories">
                        <div className="cardStyle">
                            <img src={CatBaby} className="card-img-top" alt="..." />
                        </div>
                    </button>
                </div>

                {/* New Products */}
                <div className="home-title mt-5 mb-4">
                    <h4>New Products</h4>
                </div>
                <div className="row row-cols-2 row-cols-md-4 g-3 mt-2">
                    <Card image={NewProducts} />
                    <Card image={NewProducts2} />
                    <Card image={NewProducts3} />
                    <Card image={NewProducts} />
                    <Card image={NewProducts2} />
                    <Card image={NewProducts3} />
                    <Card image={NewProducts} />
                    <Card image={NewProducts2} />
                    <Card image={NewProducts3} />
                    <Card image={NewProducts} />
                    <Card image={NewProducts2} />
                    <Card image={NewProducts3} />
                    <Card image={NewProducts} />
                    <Card image={NewProducts2} />
                    <Card image={NewProducts3} />
                    <Card image={NewProducts} />
                    <Card image={NewProducts2} />
                    <Card image={NewProducts3} />
                    <Card image={NewProducts} />
                    <Card image={NewProducts2} />

                </div>
            </div>
        </LayoutNavbarFooter>


    )
}