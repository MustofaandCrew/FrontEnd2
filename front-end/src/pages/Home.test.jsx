import {render ,screen} from "@testing-library/react";
import renderer from "react-test-renderer";

import Home from "./Home";

describe("home", () => {
    test("carousel have class carousel", async () => {
        render(<Home />);
        if (!screen.getByTestId("loading")) {
            const test = screen.getByTestId("testing");
            // eslint-disable-next-line jest/no-conditional-expect
            return (expect(test).toHaveClass("carousel"));
        }
    });

    test("snapshot", async ()=>{
        render(<Home />);
        if (screen.getByTestId("loading")) {
            const elem = renderer.create(<Home />).toJSON();
            // eslint-disable-next-line jest/no-conditional-expect
            expect(elem).toMatchSnapshot();  
        }
    });
})