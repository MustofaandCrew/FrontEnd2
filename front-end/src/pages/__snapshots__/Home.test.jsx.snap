// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home snapshot 1`] = `
<span
  data-testid="loading"
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": "100vh",
      "justifyContent": "center",
      "margin": "0 auto",
      "position": "relative",
      "width": "100%",
    }
  }
>
  <span
    style={
      Object {
        "animation": "react-spinners-PropagateLoader-propogate-0 1.5s infinite",
        "animationFillMode": "forwards",
        "background": "#FF7158",
        "borderRadius": "50%",
        "fontSize": "16.666666666666668px",
        "height": "50px",
        "position": "absolute",
        "width": "50px",
      }
    }
  />
  <span
    style={
      Object {
        "animation": "react-spinners-PropagateLoader-propogate-1 1.5s infinite",
        "animationFillMode": "forwards",
        "background": "#FF7158",
        "borderRadius": "50%",
        "fontSize": "16.666666666666668px",
        "height": "50px",
        "position": "absolute",
        "width": "50px",
      }
    }
  />
  <span
    style={
      Object {
        "animation": "react-spinners-PropagateLoader-propogate-2 1.5s infinite",
        "animationFillMode": "forwards",
        "background": "#FF7158",
        "borderRadius": "50%",
        "fontSize": "16.666666666666668px",
        "height": "50px",
        "position": "absolute",
        "width": "50px",
      }
    }
  />
  <span
    style={
      Object {
        "animation": "react-spinners-PropagateLoader-propogate-3 1.5s infinite",
        "animationFillMode": "forwards",
        "background": "#FF7158",
        "borderRadius": "50%",
        "fontSize": "16.666666666666668px",
        "height": "50px",
        "position": "absolute",
        "width": "50px",
      }
    }
  />
  <span
    style={
      Object {
        "animation": "react-spinners-PropagateLoader-propogate-4 1.5s infinite",
        "animationFillMode": "forwards",
        "background": "#FF7158",
        "borderRadius": "50%",
        "fontSize": "16.666666666666668px",
        "height": "50px",
        "position": "absolute",
        "width": "50px",
      }
    }
  />
  <span
    style={
      Object {
        "animation": "react-spinners-PropagateLoader-propogate-5 1.5s infinite",
        "animationFillMode": "forwards",
        "background": "#FF7158",
        "borderRadius": "50%",
        "fontSize": "16.666666666666668px",
        "height": "50px",
        "position": "absolute",
        "width": "50px",
      }
    }
  />
</span>
`;
